### DISCUSSION TITLE
how to deployment a cluster with docker?

### DISCUSSION THREAD

**Discussion by vCassius** [openobserve/openobserve#1403] [Q&A]
how to deployment a cluster with docker?The single-machine mode is unstable

**Comment by marksugar** [comment]
```
OpenObserve is an observability platform that allows you to capture, search, and analyze your logs, metrics, and traces.

Usage: openobserve [COMMAND]

Commands:
  reset           reset openobserve data
  view            view openobserve data
  file-list       migrate file-list from s3 to dynamo db
  delete-parquet  delete parquet files from s3 and file_list
  help            Print this message or the help of the given subcommand(s)

Options:
  -h, --help     Print help
  -V, --version  Print version

```
in k8s env ZO_NODE_ROLE
If I understand the direct relationship and configuration of these roles, that's fine. But there is no explanation.

Is there any information on docker deployment clusters? @hengfeiyang 

thanks

**Comment by prabhatsharma** [comment]
@Cassuis What do you mean by single machine mode is unstable? Do you mean it does not have HA?

What do you mean by `docker deployment cluster` ? docker compose ? We recommend, that you use Kubernetes for HA deployments if you are looking for HA deployment.

**Reply by vCassius** [reply]
do not like this . i just want to run docker and not docker compose,and i have three server, and use them run docker for HA , and it used server disk without OSS to store data

**Reply by hengfeiyang** [reply]
In cluster mode, you must have three parts:

1. etcd cluster
2. remote storage, like: S3, MinIO, OSS
3. Different Node role of OpenObserve. every node role needs to deploy at least one pod:
  1. Ingester
  2. Querier
  3. Router
  4. Compactor
  5. AlertManager 

Then you can deploy different nodes on different physical machine, but these machine have to connect each others.

**Reply by marksugar** [reply]
@hengfeiyang 
Configure different ports for different roles to communicate, so that  can run multiple role on a node. ？
Requires an example of docker-compose

thanks
